clients/clients-integration-tests/src/test/java/org/apache/kafka/clients/producer/ProducerSendWhileDeletionTest.java

```diff
@@ -157,6 +164,46 @@ public void testSendWithRecreatedTopic() throws Exception {
         }
     }
 
+    @Timeout(90)
+    @ClusterTest
+    public void testSendWhileTopicGetRecreated() throws InterruptedException {
+        int maxNumTopicRecreationAttempts = 5;
+        var recreateTopicFuture = CompletableFuture.supplyAsync(() -> {
+            var topicIds = new HashSet<Uuid>();
+            while (topicIds.size() < maxNumTopicRecreationAttempts) {
+                try (var admin = cluster.admin()) {
+                    if (admin.listTopics().names().get().contains(topic)) {
+                        admin.deleteTopics(List.of(topic)).all().get();
+                    }
+                    topicIds.add(admin.createTopics(List.of(new NewTopic(topic, 1, (short) 1))).topicId(topic).get());
+                } catch (Exception e) {
+                    // ignore
+                }
+            }
+            return topicIds;
+        });
+
+        Map<String, Object> producerConfig = Map.of(
+                ProducerConfig.INTERCEPTOR_CLASSES_CONFIG, MockProducerInterceptor.class.getName(),
+                "mock.interceptor.append", "",
+                ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, MockSerializer.class.getName(),
+                ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, MockSerializer.class.getName());
+
+        var producerFuture = CompletableFuture.runAsync(() -> {
+            try (var producer = cluster.producer(producerConfig)) {
+                for (int i = 1; i <= numRecords; i++) {
+                    producer.send(new ProducerRecord<>(topic, String.valueOf(i), ("value" + i)));
+                }
+            }
+        });
+        var topicIds = recreateTopicFuture.join();
+        producerFuture.join();
+        assertEquals(maxNumTopicRecreationAttempts, topicIds.size());
+        TestUtils.waitForCondition(() ->
+                        numRecords == MockProducerInterceptor.ON_SUCCESS_COUNT.intValue(),
```

Comment:
Hmm, this still doesn't seem quite right. We are only sending numRecords. If some of the records fail to send, ON_SUCCESS_COUNT is never going to be numRecords. 

To me, the goal of this test is to make sure that the callback for each send is called, with or w/o an exception. Perhaps we could do sth like the following?

```
           for (int i = 1; i <= numRecords; i++) {
                    var resp = producer.send(new ProducerRecord<>(topic, null, ("value" + i).getBytes()),
                            (metadata, exception) -> {
                                    numAcks.incrementAndGet();
                                }
                            });
             }
             producer.flush();

          ....

assertEquals(numRecords, numAcks.intValue());
```

Reply:
okay updated now, waiting for pipeline to pass all checks and tests